packages/redux/src/contents/utils.ts

```diff
@@ -20,6 +21,151 @@ export const ENVIRONMENT_CODES = {
 };
 
+/**
+ * Generate a ranking number for each commerce page.
+ *
+ * @function
+ *
+ * @param {object} metadata - Metadata object with custom attributes.
+ * @returns {number} Ranking number for a specific commerce page.
+ *
+ * @example
+ * const pageRanking = getPageRanking({ custom: { type: "listing" } });
+ * Result of pageRanking === 0;
+ */
+export const getPageRanking = (metadata: Metadata): number => {
+  let ranking = 0;
+  const { gender, brand, category, priceType, id } = metadata.custom;
+
+  // Return default ranking number when no data was returned from each commerce property.
+  // We can't only check if it's empty from 'metadata.custom' because it could have other properties inside.
+  if (!gender && !brand && !category && !priceType && !id) {
+    return ranking;
+  }
+
+  const rankingGender = 10;
+  const rankingBrand = 100;
+  const rankingCategory = 10000;
+  const rankingPriceType = 1000000000;
+  const rankingId = 10000000000;
+
+  Object.entries(metadata.custom).forEach(entry => {
+    const [key, value] = entry;
+
+    if (isEmpty(value)) {
+      return ranking;
+    }
+
+    switch (key) {
+      case 'gender':
+        ranking += rankingGender;
+        break;
+      case 'brand':
+        ranking += rankingBrand;
+        break;
+      case 'category':
+        ranking += rankingCategory;
+        break;
+      case 'priceType':
+        ranking += rankingPriceType;
+        break;
+      case 'id':
+        ranking += rankingId;
+        break;
+
+      default:
+        return ranking;
+    }
+  });
+
+  return ranking;
+};
+
+/**
+ * Method to check each page ranking and return the better one.
+ *
+ * @function
+ *
+ * @param {object} result - Commerce page payload result.
+ * @returns {object} Selected page with better ranking.
+ *
+ * @example
+ * const defaultStrategy = getDefaultStrategy({ entries: [...pages] });
+ * Result of defaultStrategy === { entries: [...bestRankedPage] };
+ */
+export const getDefaultStrategy = (
+  result: CommercePagesContentNormalized,
+): CommercePagesContentNormalized => {
+  const bestRankedPage = result.entries
+    .map(page => ({
+      ...page,
+      ranking: getPageRanking(page.metadata),
+    }))
+    .reduce((p, c) => (p.ranking > c.ranking ? p : c));
+
+  return {
+    number: 1,
+    totalPages: 1,
+    totalItems: 1,
+    entries: [bestRankedPage],
+  };
+};
+
+/**
+ * Method to return the main page with all components from other pages.
+ *
+ * @function
+ *
+ * @param {object} result - Commerce page payload result.
+ * @returns {object} Selected page with merged components.
+ *
+ * @example
+ * const mergeStrategy = getMergeStrategy({ entries: [...pages] });
+ * Result of mergeStrategy === { entries: [...mergedPages] };
+ */
+export const getMergeStrategy = (
+  result: CommercePagesContentNormalized,
+): CommercePagesContentNormalized => {
+  // Merge components inside the selected commerce page
+  const mergedComponents = [];
+  result.entries.map(content =>
```

Comment:
This is more a nitpick than anything else, but what about replacing `.map` with `forEach`? Since you're not storing the array the map returns, a `forEach` seems more adequate 🙂  (same for the line below)

And please add an empty line above 😇

Reply:
Not nitpick at all, forEach is better here.